app/services/school_partnerships/create.rb

```diff
@@ -0,0 +1,119 @@
+# TODO: error messages are currently consistent with ECF, however we want them to use RECT
+# terminology in the service and will look to map between them somehow in a future PR.
+module SchoolPartnerships
+  class Create
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :lead_provider_id
+    attribute :contract_period_year
+    attribute :school_api_id
+    attribute :delivery_partner_api_id
+
+    validates :contract_period_year, presence: { message: "Enter a '#/cohort'." }
+    validates :school_api_id, presence: { message: "Enter a '#/school_id'." }
+    validates :lead_provider_id, presence: { message: "Enter a '#/lead_provider_id'." }
+    validates :delivery_partner_api_id, presence: { message: "Enter a '#/delivery_partner_id'." }
+    validate :enabled_contract_period_year_exists
+    validate :lead_provider_exists
+    validate :school_exists
+    validate :school_is_not_cip_only
+    validate :school_is_eligible
+    validate :delivery_partner_exists
+    validate :lead_provider_delivery_partnership_exists
+    validate :school_partnership_does_not_already_exists
+    validate :not_school_led
+
+    def create
+      return false unless valid?
+
+      SchoolPartnership.create!(
```

Comment:
I think we'll want to [record an event here](https://github.com/DFE-Digital/register-early-career-teachers-public/blob/main/app/services/events/record.rb) too.

Reply:
Added 👍 we don't have any LP events yet so I added a new user type for them.

Reply:
Ooo I need to start adding those to action API tickets 👍